src/Sulu/Component/Security/Authorization/AccessControl/DescendantProviderInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Security\Authorization\AccessControl;
+
+interface DescendantProviderInterface
+{
+    /**
+     * Return the children for access control inherit update.
+     *
+     * @param string|int $id
+     *
+     * @return string[]|int[]
+     */
+    public function getDescendantIdsById($id);
```

Comment:
Arent't these methods mostly named `find...`?

Reply:
Is okay for me. It was just once in the past where we accidentally conflicted with a doctrine function and so the interface combination of our and doctrine repository interface did not longer work, thats why I'm also a little bit worried when naming a function in a repository interface with find 🙈

Reply:
Okay, I see... but seems to be highly unlikely in that case, right? Also saw a talk once, where people did not even use the doctrine repository as the base class :see_no_evil:

Reply:
@danrot In our new projects we did that without extending but in the latest release of doctrine they added sadly a typehint so now all registred repositories need to extend from ObjectRepository else it will fail. Did not had a deeper look at it as @nnatter / @luca-rath did run into that issue and then changed to extend from the doctrine repository again.

Reply:
But I guess they will only fail if you register them in Doctrine as `repositoryClass` of an entity, right? If we only use Symfony's dependency injection to get the repositories that should not really matter, does it? Just curious, it obviously is out of scope of this PR anyway :see_no_evil: